textarea/elfinder-sakai/src/java/org/sakaiproject/elfinder/sakai/SakaiFsService.java

```diff
@@ -179,6 +181,10 @@ public String getVolumeId(FsVolume volume) {
 			if("!admin".equals(currentSiteId) || "~admin".equals(currentSiteId)){
 				continue;
 			}
+			//Exclude a site where the user does not have access because it's secure.
+			if(!securityService.unlock(siteService.SITE_VISIT, "/site/"+currentSiteId)){
```

Comment:
Calling `siteService.allowGetSite(site)` would seem to be the better API to call here.